janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java

```diff
@@ -0,0 +1,169 @@
+     * graphName or set the storage directory to the storage_root + /graphName.</p>
+     *
+     * @param graphName
+     *
+     * @return JanusGraph
+     */
+    public static JanusGraph open(String graphName) {
+        final ConfigurationManagementGraph configGraphManagement = getConfigGraphManagementInstance();
+        final Map<String, Object> graphConfigMap = configGraphManagement.getConfiguration(graphName);
+        Preconditions.checkState(null != graphConfigMap,
+                                "Please create configuration for this graph using the ConfigurationManagementGraph#createConfiguration API.");
+        JanusGraphManager jgm = JanusGraphManager.getInstance();
+        Preconditions.checkState(jgm != null, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
+        return (JanusGraph) jgm.openGraph(graphName, (String gName) -> {
+            return new StandardJanusGraph(new GraphDatabaseConfiguration(new CommonsConfiguration(new MapConfiguration(graphConfigMap))));
+        });
+    }
+
+    /**
+     * Get a Set of the graphNames that exist in your configuration management graph
+     *
+     */
+    public static Set<String> getGraphNames() {
+        final ConfigurationManagementGraph configGraphManagement = getConfigGraphManagementInstance();
+        final List<Map<String, Object>> configurations = configGraphManagement.getConfigurations();
+        return configurations.stream().map(elem -> {
+            return (String) elem.getOrDefault(ConfigurationManagementGraph.PROPERTY_GRAPH_NAME, null);
+        }).filter(Objects::nonNull).collect(Collectors.toSet());
+    }
+
+    /**
+     * Removes the graph corresponding to the supplied {@link String} graphName
+     * from the {@link JanusGraphManager} {@link Map<String, Graph>} graph reference tracker and
+     * returns the corresponding Graph, or null if it doesn't exist.
+     *
+     * @param configuration Graph
+     * @return JanusGraph
+     */
+    public static JanusGraph close(String graphName) throws Exception {
+        final JanusGraphManager jgm = JanusGraphManager.getInstance();
+        Preconditions.checkState(jgm != null, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
+        Graph graph = jgm.removeGraph(graphName);
+        if (null != graph) graph.close();
+        return (JanusGraph) graph;
+    }
+
+    /**
+     * Returns a {@link Configuration} corresponding to supplied .properties file
+     * which lives at the supplied {@link String} fileLocation.
+     *
+     * @param {@link String} properties file location
+     *
+     * @return Configuration
+     */
+    public static Configuration fileToConfiguration(final String fileLocation) throws IOException, ConfigurationException {
+        return new PropertiesConfiguration(fileLocation);
+    }
+
+    private static ConfigurationManagementGraph getConfigGraphManagementInstance() {
+        ConfigurationManagementGraph configGraphManagement = null;
```

Comment:
Just don't set it to null initially and it's valid since you propagate a RuntimeException in the catch.

```java
final ConfigurationManagementGraph configGraphManagement;
try {
    configGraphManagement = ConfigurationManagementGraph.getInstance();
} catch (ConfigurationManagementGraphNotEnabledException e) {
    throw new RuntimeException(e);
}
return configGraphManagement;
```